src/constructs/cloudwatch/api-gateway-alarms.ts

```diff
@@ -0,0 +1,45 @@
+import { Duration } from "aws-cdk-lib";
+import type { RestApi } from "aws-cdk-lib/aws-apigateway";
+import { ComparisonOperator, MathExpression, TreatMissingData } from "aws-cdk-lib/aws-cloudwatch";
+import type { GuStack } from "../core";
+import { AppIdentity } from "../core";
+import type { GuAlarmProps, Http5xxAlarmProps } from "./alarm";
+import { GuAlarm } from "./alarm";
+
+interface GuApiGateway5xxPercentageAlarmProps
+  extends Pick<GuAlarmProps, "snsTopicName">,
+    Http5xxAlarmProps,
+    AppIdentity {
+  apiGatewayInstance: RestApi;
+}
+
+/**
+ * Creates an alarm which is triggered whenever the percentage of requests with a 5xx response code exceeds
+ * the specified threshold.
+ */
+export class GuApiGateway5xxPercentageAlarm extends GuAlarm {
+  constructor(scope: GuStack, props: GuApiGateway5xxPercentageAlarmProps) {
+    const mathExpression = new MathExpression({
+      expression: "100*m1/m2",
+      usingMetrics: {
+        m1: props.apiGatewayInstance.metricServerError(),
+        m2: props.apiGatewayInstance.metricCount(),
+      },
+      label: `% of 5XX responses served for ${props.app}`,
+      period: Duration.minutes(1),
+    });
+    const defaultAlarmName = `High 5XX error % from ${props.app} (ApiGateway) in ${scope.stage}`;
+    const defaultDescription = `${props.app} exceeded ${props.tolerated5xxPercentage}% error rate`;
+    const alarmProps = {
+      ...props,
+      metric: mathExpression,
+      treatMissingData: TreatMissingData.NOT_BREACHING,
+      threshold: props.tolerated5xxPercentage,
+      comparisonOperator: ComparisonOperator.GREATER_THAN_THRESHOLD,
+      alarmName: props.alarmName ?? defaultAlarmName,
```

Comment:
minor: Could we use destructuring with default values (e.g [here](https://github.com/guardian/cdk/blob/b140d9f4025ebf56eb6b0cd2279e6e18a551eed1/src/constructs/ecs/ecs-task.ts#L142-L157)) to improve readability a little?

Reply:
I had a quick look at this but destructuring makes things awkward here as we just want to pass the whole object into the [superclass](https://github.com/guardian/cdk/blob/8b434a5d728cb9fbcd73dbf3645927b50e4afb93/src/constructs/cloudwatch/api-gateway-alarms.ts#L43).